xbmc/addons/AddonInstaller.cpp

```diff
@@ -207,6 +207,15 @@ bool CAddonInstaller::InstallOrUpdateDependency(const ADDON::AddonPtr& dependsId
                    DependencyJob::YES, AllowCheckForUpdates::YES);
 }
 
+bool CAddonInstaller::RemoveDependency(const std::shared_ptr<IAddon>& dependsId) const
+{
+  bool removeData = CDirectory::Exists("special://profile/addon_data/" + dependsId->ID());
+  CAddonUnInstallJob removeDependencyJob(dependsId, removeData);
+  removeDependencyJob.SetRecurseOrphaned(RecurseOrphaned::NO);
+
+  return removeDependencyJob.DoWork();
+}
```

Comment:
imo `removeDependencyJob.DoWork()` is correct here as we should not show a progress indicator. we cannot know how many deps will get purged beforehand.

Reply:
@Montellese what‘s your opinion on this? i think this section was your work long time ago